libs/icons/src/components/sage-icon/sage-icon.tsx

```diff
@@ -0,0 +1,151 @@
+import { Build, Component, Element, Host, Prop, State, Watch, h } from '@stencil/core';
+import { getSvgContent, sageIconContent } from './request';
+import { getName, getUrl, inheritAttributes } from './utils';
+
+@Component({
+  tag: 'sage-icon',
+  assetsDirs: ['svg'],
+  styleUrl: 'sage-icon.scss',
+  shadow: true,
+})
+export class SageIcon {
+  private io?: IntersectionObserver;
+  private inheritedAttributes: { [k: string]: any } = {};
+
+  @Element() el!: HTMLElement;
+
+  @State() private ariaLabel?: string;
+  @State() private isVisible = false;
+  @State() private svgContent?: string;
+
+  /**
+   * The color to be use for the background
+   */
+  @Prop() color?: string;
+
+  /**
+   * The name of the icon to use from
+   * the built-in set.
+   */
+  @Prop({ reflect: true }) name?: string;
```

Comment:
Would be great to get a populated list of icon names here

Reply:
That's more of a feature implementation for Storybook not the component.